src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1542,6 +1542,27 @@ private void visitTypeDeclaration(JSType type, boolean isVarArgs, boolean isOpti
       }
     }
 
+    private boolean emitNoResolvedType(NoType type, String defaultEmit) {
+      if (!opts.partialInput || !type.isNoResolvedType()) {
+        emit(defaultEmit);
+        return true;
+      }
+      // When processing partial inputs, this case handles implicitly forward declared types
+      // for which we just emit the literal type written along with any type parameters.
+      NoResolvedType nType = (NoResolvedType) type;
+      if (nType.getReferenceName() != null) {
+        String displayName = type.getDisplayName();
+        emit(Constants.INTERNAL_NAMESPACE + "." + displayName);
+        List<JSType> templateTypes = nType.getTemplateTypes();
+        if (templateTypes != null && templateTypes.size() > 0) {
+          emitGenericTypeArguments(type.getTemplateTypes().iterator());
+        }
+      } else {
```

Comment:
I think I'd invert the if and do another early return here.

Reply:
done